Fixes navigation bug in InventoryAdd Adds SCM Branch field on JTForm

AlexSCorey opened the pull request:
##### SUMMARY
This PR addresses #5619 and #5652.  It adds the SCM_Branch to JT form if the user selects a project with `allow_override: true`.  To do this I exposed the `allow_override` boolean inside the `job_template.summary_fields.projects`

It also gives a navigation error after adding an Inventory.  With this PR the user is redirected to the details field correctly.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - API
 - UI

##### AWX VERSION
```
9.1.0
```


##### ADDITIONAL INFORMATION
In this gif the Project 10 Org 2 has override enabled
![SCM Branch](https://user-images.githubusercontent.com/39280967/72366962-50ca8f00-36c9-11ea-864b-a53604c28297.gif)

![Inventory Form Navigation](https://user-images.githubusercontent.com/39280967/72366968-54f6ac80-36c9-11ea-852c-3f515f7d5459.gif)

AlexSCorey committed:
Fixes navigation bug in InventoryAdd Adds SCM Branch field on JTForm

AlexSCorey committed:
Fixes navigation bug in InventoryAdd Adds SCM Branch field on JTForm

jlmitch5 commented on awx/api/serializers.py:
```diff
@@ -113,7 +113,7 @@
                                        'groups_with_active_failures',
                                        'has_inventory_sources'),
-    'project': DEFAULT_SUMMARY_FIELDS + ('status', 'scm_type'),
+    'project': DEFAULT_SUMMARY_FIELDS + ('status', 'scm_type', 'allow_override'),
```
this is cool, and smarter than the way I did it in the legacy ui (calling the project detail endpoint to get the info when project changes).  fantastic stuff!

AlanCoding replied:
Since `scm_type` is already a property of the Project subclass, there should be no additional API database queries from adding this field.

👍

AlexSCorey replied:
Per offline convo with @matburt I am going to undo this and go a different route that does not require changes to the api.

jlmitch5 commented on awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx:
```diff
@@ -590,6 +601,7 @@ const FormikApp = withFormik({
       job_type: template.job_type || 'run',
       inventory: template.inventory || '',
       project: template.project || '',
+      scm_branch: template.scm_branch || '',
```
I'm not sure if the api will complain if you pass branch (even if it's empty string) if allow_override is false for the project.  need to pull down and test that

AlexSCorey replied:
@jlmitch5  It does complain, but before I submit that info I check to see if the project.allow_override:true. If it is not, then I remove that scm_branch value.

jlmitch5 replied:
nice

jlmitch5 commented on awx/ui_next/src/screens/Template/shared/JobTemplateForm.test.jsx:
```diff
@@ -150,9 +152,18 @@ describe('<JobTemplateForm />', () => {
+      target: { value: 'devel', name: 'scm_branch' },
+    });
+    expect(form.state('values').scm_branch).toEqual('devel');
+
```
nice test

jlmitch5 reviewed (commented):
code looks great, pulling down now

AlexSCorey committed:
Removes code from serializer in favor to api call of Project.readDetails

Adds necessary tests.

AlexSCorey committed:
Removes code from serializer in favor to api call of Project.readDetails

Adds necessary tests.

AlexSCorey committed:
Resets playbook and scm-branch fields when project is changed

The playbook field becomes undefined and the scm-branch
field becomes ''.  This ensures that the user has to assign
a playbook to the template that is associated with the project
and suggests to the user to review their scm-branch.
TODO: when the user updates project with scm-branch override
allow the user to type in playbook in dropdown. Then, check if
playbook is present in list of playbooks.  If no, add it to the
list of playbooks.

jlmitch5 reviewed (approved):
awesome looks good!
